### PR TITLE
Dictionary example

### PR DESCRIPTION
**Pull request by xsebek**
Add an example dictionary and set implemented as a red-black tree, based on [Persistent Red Black Trees in Haskell](https://abhiroop.github.io/Haskell-Red-Black-Tree/).

### CODE REVIEW

File: example/dictionary.sw [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+      }
+    )
+  in if (t.c == red) {
+    case5 t
+  } {
+    balanceL t
+  }
+end
+
+def insertT : (k -> a) -> k -> RBTree k -> RBTree k = \p.\x.\t.
```
Comment (xsebek) [inline example/dictionary.sw:135]: There is a bug somewhere in insert or balance:
```
> def i = insertS end
> def s = i 7 $ i 8 $ i 32 $ i 52 $ i 62 $ i 76 $ i 80 emptyS end
> formatS s
it4 : Text = "[7, 8, 32, 52, 62, 76, 80]"
> formatS $ i 54 s
it5 : Text = "[7, 8, 32, 62, 54, 62, 76, 80]"
```

| S | S + 54 |
|--------|--------|
| Screenshot 2024-10-14 at 10 44 19 PM | Screenshot 2024-10-14 at 10 44 42 PM |

Comment (xsebek) [reply example/dictionary.sw:154]: Fixed! There was a tree duplication in balance in two places caused by copy-paste.

**Review by byorgey — APPROVED** [review]
This is incredible! :evergreen_tree: :tada:

**Comment by xsebek** [comment]
I have some ideas for improvement, but I will leave them for another PR. 🙂

**Pull request merged**